addons/stock_landed_costs/models/stock_landed_cost.py

```diff
@@ -78,12 +78,15 @@ def button_cancel(self):
     def button_validate(self):
         if any(cost.state != 'draft' for cost in self):
             raise UserError(_('Only draft landed costs can be validated'))
-        if any(not cost.valuation_adjustment_lines for cost in self):
-            raise UserError(_('No valuation adjustments lines. You should maybe recompute the landed costs.'))
-        if not self._check_sum():
-            raise UserError(_('Cost and adjustments lines do not match. You should maybe recompute the landed costs.'))
 
         for cost in self:
+            if not cost.picking_ids:
+                raise UserError(_('Please define the transfers on which those additional costs should apply.'))
+            if not cost.valuation_adjustment_lines:
+                cost.compute_landed_cost()
```

Comment:
Could you call compute_landed_cost on self directly ?

Reply:
nope, as the user could manually change the generated lines before validating the lc

Reply:
ok i could try to batch a little more